src/wasm/wasm-s-parser.cpp

```diff
@@ -3209,13 +3202,37 @@ void SExpressionWasmBuilder::parseTable(Element& s, bool preParseImport) {
       throw ParseException("invalid table", inner.line, inner.col);
     }
   }
-  if (i == s.size()) {
-    wasm.addTable(std::move(table));
-    return;
+
+  bool hasExplicitLimit = false;
+
+  if (s[i]->isStr()) {
+    char c = s[i]->c_str()[0];
+    if ('0' <= c && c <= '9') {
```

Comment:
You can use `wasm::isInteger` to check for numbers here and below.

Reply:
I'm not sure if I understand the point here, `wasm::isInteger` seems to check that a `double` is actually an integer and also not infinite.

Reply:
Sorry, I misremembered the function name. I meant `String::isNumber` :)

Reply:
Oh nice! Thanks. Fixed now. By the way, do we parse hex numbers in general? Is there a utility for detecting and parsing hex the way it's defined and used in the spec? e.g. `0x1000_0000`

Reply:
No, not that I know of.